### PR TITLE
Integration Test: Startup

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
- A new section called `Testing` in the PR template. This is to encourage people to consider whether a change needs to have a test case added.
- New Integration test, the startup test. This test utilizes IMO K3s's best feature: its fast startup time, to test a variety of startup arguments and scenarios. The case structure is incredible simple, k3s starts up, checks if basic pods are deployed correctly, kills itself. The vision is enable other developers to quickly make future test cases as new bugs and edge cases around startup flags are fixed. 
- Underlying change from CLI string parsing of "kubectl" output to internal k8s client calls. Conforms more to standard k8s and allows us to use standard k8s structs.

#### Types of Changes ####
Testing Improvements

#### Verification ####
All integration tests pass... in about 8.5 minutes locally (6 cores, 12threads)

#### Linked Issues ####
N/A Internal testing change only

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: tests/integration/integration.go [hunk 1]
```diff
@@ -120,6 +138,92 @@ func K3sServerArgs() []string {
...
+	}
+	for d, found := range deploymentSet {
+		if !found {
+			return fmt.Errorf("failed to deploy %s\n%s", d, res)
+		}
+	}
+	return nil
+}
+
+func ParsePods() ([]Pod, error) {
```
Comment (brandond) [inline tests/integration/integration.go:174]: Is now maybe a good time to come up with some helpers to create an actual Kubernetes client and just get the nodes and pods directly, instead of wrapping shell commands and parsing the output into our own Pod/Node structs?

Comment (dereknola) [reply tests/integration/integration.go]: Resolved

**Review by brandond — APPROVED** [review]
